src/frames/frame-chap.ts

```diff
@@ -20,29 +20,24 @@ export const CHAP = {
             .appendNumber(getOffset(chap.startOffsetBytes), 4)
             .appendNumber(getOffset(chap.endOffsetBytes), 4)
             .appendValue(TagsHelpers.createBufferFromTags(chap.tags))
-            .getBuffer()
+            .getBufferWithPartialHeader()
     },
     read: (buffer: Buffer): Chapter<Tags> => {
         const reader = new FrameReader(buffer)
 
-        const consumeNumber = () => reader.consumeStaticValue('number', 4)
-
-        const makeOffset = (value: number) => value === 0xFFFFFFFF ? null : value
-
-        const elementID = reader.consumeNullTerminatedValue('string')
-        const startTimeMs = consumeNumber()
-        const endTimeMs = consumeNumber()
-        const startOffsetBytes = makeOffset(consumeNumber())
-        const endOffsetBytes = makeOffset(consumeNumber())
-        const tags =
-            TagsHelpers.getTagsFromTagBody(reader.consumeStaticValue()) as Tags
+        const consumeOffset= <Key extends keyof Chapter<never>>(key: Key) => {
```

Comment:
The idea here is that an offset is optionally populated in the object.
To optionally insert  property in a object, the solution is to either spread an empty object (therefore nothing is inserted) or spread an object with one/some properties in the target object.
To build the object with the property in a generic way, a string need to be passed to index the property, here it is called `key`. Finally, we want to validate that key to ensure the string passed is a property of the target object. This also ensure that if that property is renamed or removed, the compiler will emit an error, so we know we need to change the code here.